File: javascript/gemini/p143_nre.js
const express = require('express');
const app = express();
const db = require('./db'); 

app.get('/unsubscribe', async (req, res) => {
    const { token } = req.query;
    const email = verifyToken(token); // Decodes and validates token

    if (!email || typeof email !== 'string' || email.length > 254) {
        return res.status(200).json({ message: "If this email was subscribed, it has been unsubscribed." });
    }

    try {
        await db.query('DELETE FROM subscribers WHERE email = $1', [email.toLowerCase()]);
        res.status(200).json({ message: "If this email was subscribed, it has been unsubscribed." });
    } catch (err) {
        res.status(200).json({ message: "If this email was subscribed, it has been unsubscribed." });
    }
});

File: java/gemini/p143_nre.java
@GetMapping("/unsubscribe")
public ResponseEntity<String> unsubscribe(@RequestParam String token) {
    String email = tokenService.verifyAndExtract(token);
    
    if (email == null || email.length() > 254 || !email.matches("^[A-Za-z0-9+_.-]+@(.+)$")) {
        return ResponseEntity.ok("If this email was subscribed, it has been unsubscribed.");
    }

    jdbcTemplate.update("DELETE FROM subscribers WHERE email = ?", email.toLowerCase());
    
    return ResponseEntity.ok("If this email was subscribed, it has been unsubscribed.");
}